packages/alfa-dom/src/tables/cell.ts

```diff
@@ -0,0 +1,703 @@
+    y: number,
+    w: number,
+    h: number,
+    element: Element,
+    headers: Array<Element> = []
+  ): Cell {
+    return new Cell(kind, x, y, w, h, element, headers);
+  }
+
+  private constructor(
+    kind: Cell.Kind,
+    x: number,
+    y: number,
+    w: number,
+    h: number,
+    element: Element,
+    headers: Array<Element>
+  ) {
+    this._kind = kind;
+    this._anchorX = x;
+    this._anchorY = y;
+    this._width = w;
+    this._height = h;
+    this._element = element;
+    this._headers = headers;
+  }
+
+  // debug
+  public get name(): string {
+    return this._element.attribute("id").get().value;
+  }
+
+  public get anchor(): { x: number; y: number } {
+    return { x: this._anchorX, y: this._anchorY };
+  }
+  public get width(): number {
+    return this._width;
+  }
+  public get height(): number {
+    return this._height;
+  }
+  public get kind(): Cell.Kind {
+    return this._kind;
+  }
+  public get element(): Element {
+    return this._element;
+  }
+  public get headers(): Iterable<Element> {
+    return this._headers;
+  }
+
+  public isCovering(x: number, y: number): boolean {
+    return !(
+      // cell is *not* covering if either
+      (
+        x < this._anchorX || // slot is left of cell
+        this._anchorX + this._width - 1 < x || // slot is right of cell
+        y < this._anchorY || // slot is above cell
+        this._anchorY + this._height - 1 < y
+      ) // slot is below cell
+    );
+  }
+
+  /**
+   * compare cell according to their anchor
+   * in a given group of cells (row, rowgroup, table, …), no two different cells can have the same anchor, so this is good.
+   */
+  public compare(cell: Cell): number {
+    if (this._anchorY < cell._anchorY) return -1;
+    if (this._anchorY > cell._anchorY) return 1;
+    if (this._anchorX < cell._anchorX) return -1;
+    if (this._anchorX > cell._anchorX) return 1;
+    return 0;
+  }
+
+  public equals(value: unknown): value is this {
+    return (
+      value instanceof Cell &&
+      this._kind === value._kind &&
+      this._width === value._width &&
+      this._height === value._height &&
+      this._anchorX === value._anchorX &&
+      this._anchorY === value._anchorY &&
+      this._element.equals(value._element)
+    );
+  }
+
+  public toJSON(): Cell.JSON {
+    return {
+      kind: this._kind,
+      anchor: this.anchor,
+      width: this._width,
+      height: this._height,
+      element: this.name, // this._element.toJSON(),
+      headers: this._headers.map(
+        (header) => header.attribute("id").get().value
+      ), // this._headers.map((header) => header.toJSON()),
+    };
+  }
+}
+
+export namespace Cell {
+  export interface JSON {
+    [key: string]: json.JSON;
+
+    kind: Kind;
+    anchor: { x: number; y: number };
+    width: number;
+    height: number;
+    element: string; //Element.JSON;
+    headers: string[]; //Element.JSON[];
+  }
+
+  export enum Kind {
+    Header = "HEADER",
+    Data = "DATA",
```

Comment:
```suggestion
    Header = "header",
    Data = "data",
```

All string constants in Alfa are kebab cased.